pkg/proxy/proxy.go

```diff
@@ -488,10 +489,17 @@ func (p *Proxy) CreateOrUpdateRedirect(l4 *policy.L4Filter, id string, source Pr
 		return nil, err
 	}
 
-	if strings.ToLower(l4.L7Parser) != "http" {
+	if !(strings.ToLower(l4.L7Parser) == "http" ||
```

Comment:
You should never have to convert to lowercase, and this can be made a little cleaner:
```
switch l4.L7Parser {
case "http":
case "kafka":
    log.Debug("MK in CreateOrUpdateRedirect l4.L7Parser:..returning", l4.L7Parser)
    return nil, fmt.Errorf("unsupported L7 protocol proxy:\"%s\"", l4.L7Parser)
default:
    return nil, fmt.Errorf("unknown L7 protocol \"%s\"", l4.L7Parser)
}

Reply:
done